### ISSUE TITLE
`LoginWithShortLivedTokenPage` needs improvement

### ISSUE THREAD

**Issue by marcaaron** [Expensify/App#8229]
## Problem

The transition page logic feels like it is a bit brittle and difficult to understand. There are not obvious side effects like performing some actions in `componentDidUpdate()` which will only happen if we connect to the `betas` because they load and update via AuthScreens. Logic like that is pretty hard to follow and therefore easy to break. 

## Why is this important?

This flow is critical to allowing users to move smoothly from OldDot to NewDot.

## Solution

Think about how to improve the code (refactor), cover, and re-test the various scenarios. Maybe write automated tests for this flow if at all possible (not sure if it is).

**Comment by marcaaron** [comment]
I know @neil-marcellini is looking into fixes for various issues. 
I'm not sure if we're doing a total refactor of the code at this point though.

**Comment by neil-marcellini** [comment]
I was able to get all of the flows to work on dev. I have this PR up for review now, [Prevent "Continue setup" navigating back immediately](https://github.com/Expensify/App/pull/8498) and these two PRs which will follow to complete the fix on the feature branch.

[[HOLD 8498] Only create 1 free workspace when transitioning](https://github.com/Expensify/App/pull/8611)
[[HOLD 8611] Wait for Onyx to clear before signing in the transitioning user](https://github.com/Expensify/App/pull/8736)

**Comment by neil-marcellini** [comment]
Oh yeah, thanks @michaelhaxhiu. I'm waiting on 1 more approval for this PR and then it will be fixed https://github.com/Expensify/App/pull/8855.

**Comment by marcaaron** [comment]
Awesome! Thanks for all your hard work and patience with this one @neil-marcellini
